src/core/server/http/http_server.ts

```diff
@@ -164,7 +164,16 @@ export class HttpServer {
         const { authRequired, tags, body = {}, timeout } = route.options;
         const { accepts: allow, maxBytes, output, parse } = body;
         // Hapi does not allow timeouts on payloads to be specified for 'head' or 'get' requests
-        const payloadTimeout = isSafeMethod(route.method) || timeout == null ? undefined : timeout;
+        const payloadTimeout = isSafeMethod(route.method) ? undefined : timeout?.payload;
+
+        let socketTimeout: number | undefined;
+        if (timeout?.idleSocket) {
+          socketTimeout = timeout.idleSocket;
+        } else if (payloadTimeout) {
+          // Hapi server requires the socket to be greater than payload settings so we add 1 millisecond
```

Comment:
It's rather confusing that `socketTimeout` is completely ignored. a couple of options:
- use a higher value `Math.max(socketTimeout, payloadTimeout + 1)`
- do not rely on Hapi and set `socket.setTimeout(...)` in a request interceptor manually

Reply:
> It's rather confusing that socketTimeout is completely ignored

If the route specifies the `timeout.idleSocket` it will [always be used](https://github.com/elastic/kibana/blob/2c939ea61d580d23d5df606c8b795f9f91380558/src/core/server/http/http_server.test.ts#L1160). Even if it leads to an error being thrown if [it's smaller than the payload timeout](https://github.com/elastic/kibana/blob/2c939ea61d580d23d5df606c8b795f9f91380558/src/core/server/http/http_server.test.ts#L1225)... 

> use a higher value Math.max(socketTimeout, payloadTimeout + 1)

If the route definition tries to explicitly set the idle socket timeout to be smaller than the payload timeout, it feels less surprising to consumers to just throw an explicit error.

> do not rely on Hapi and set socket.setTimeout(...) in a request interceptor manually

I'm hopeful that this will be a temporary requirement, and https://github.com/hapijs/hapi/pull/4123 will merge shortly. The only route that I'm aware of that will need to set the idle socket timeout is Fleet, which is adjusting the timeout to be super long to support long-polling, so they won't hit us leaking HapiJS constraints.

Reply:
Do you know if hapijs/hapi#4123 is going to be backported to Hapi v17?

Reply:
> Do you know if hapijs/hapi#4123 is going to be backported to Hapi v17?

I don't... I just asked for clarification in https://github.com/hapijs/hapi/pull/4123#issuecomment-669933397.

Reply:
Per https://github.com/hapijs/hapi/pull/4123#issuecomment-669935812

> If this lands, it will only be for v20. If you need support for an older version, you'll have to check on a commercial license - #4114.

Reply:
https://github.com/elastic/kibana/pull/73730/commits/3d8901acab2c24c873b915793c9b4ce2b65bd922 works around the HapiJS constraints by setting a socket timeout on the route definition when a payload timeout is specified, and then uses the `onPreAuth` lifecycle event to set the actual socket timeout... It's not possible to do so in the `onRequest` lifecycle event because the route hasn't been looked up yet per https://hapi.dev/api/?v=19.2.0#request-lifecycle